File: app-java/src/test/java/tech/thdev/java_udemy_sample/null_safety_02/_03_NullPointExceptionTest.java
package tech.thdev.java_udemy_sample.null_safety_02;

import org.junit.Test;

/**
 * Created by tae-hwan on 10/7/16.
 * <p>
 * NullPointException을 발생시키는 방법
 */

public class _03_NullPointExceptionTest {

    @Test
    public void test() throws InterruptedException {
        String temp = null;
        int size = temp.length();
        System.out.println(size);
    }
}

File: app-kotlin/src/test/java/tech/thdev/kotlin_udemy_sample/null_safety_02/_03_NullPointExceptionTest.kt
package tech.thdev.kotlin_udemy_sample.null_safety_02

import org.junit.Test

/**
 * Created by tae-hwan on 10/6/16.
 *
 * NullPointException을 발생시키는 방법
 */
class _03_NullPointExceptionTest {

    @Test
    @Throws(InterruptedException::class)
    fun testNullPointException() {
        var temp: String? = null
        val size = temp?.length ?: throw NullPointerException("temp is null")
        print(size)
    }

    @Test
    @Throws(InterruptedException::class)
    fun testNullPointException2() {
        var temp: String? = null
        val size = temp!!.length
        print(size)
    }
}